tests/type_definitions/main.ts

```diff
@@ -0,0 +1,7 @@
+/// <deno-types path="./foo.d.ts" />
+import * as foo from "./foo.js";
+
+/// <deno-types path="./bar.d.ts" />
+import * as bar from "./bar.js";
```

Comment:
Wouldn't it be nicer to be able to declare this in bar.js ?

Reply:
That assumes you have control of `bar.js` which I didn't think would be a common workload.  Also, in order for the compiler to see that it has to resolve and load the `.js` file.  In this case, the `.js` file is never loaded by the compiler, the `.d.ts` is instead to give TypeScript the types.

The typical use case is going to be, in my opinion, "I am using an external library named "foo", that library has types at `@types/foo`, I want to use that library in Deno, but I also want that code to work other places".  I will write something like this, likely in something like `deps.ts`:

```js
//# types=https://unpkg.com/@types/foo@~1.0.0/index.d.ts
import * as foo from "https://unpkg.com/foo@~1.0.0/index.js";
```

If you have mixed mode Deno project though, you don't need this feature, as `allowJs` is on by default, which allows you to resolve `.js` files and import them into Deno, and then `checkJs` would allow you to type check the files based on the JavaScript itself, plus and JSDoc comments.  (I also believe `// @ts-check` in each JavaScript file would also work, though I haven't tested it).

Reply:
> That assumes you have control of bar.js

Right. I see now.

In typescript currently this is done with the `<reference type="blah"/>` directive? The `//#` syntax now looks strange since it's not applying to the current file.

Reply:
Sort of...

From the TypeScript documentation:

> For example, including `/// <reference types="node" />` in a declaration file declares that this file uses names declared in `@types/node/index.d.ts;` and thus, this package needs to be included in the compilation along with the declaration file.

It is more of a directive for `.d.ts` files to say "I also need these types" and relies upon the "magical" resolution within `node_modules`.  We might actually have a bit of a problem with that now that you mention it, but at least this would be a step forward and we can figure out how to address that in the future.

I agree it does look funny, the only other suggestion, is that there are a couple TypeScript compiler "hints" like `@ts-check` and `@ts-nocheck` and `@ts-ignore`.  A "hint" style syntax might work which could be something like:

```ts
// @deno-types="https://unpkg.com/@types/foo@~1.0.0/index.d.ts"
import * as foo from "https://unpkg.com/foo@~1.0.0/index.js";
```